### PR TITLE
Update DBProfileDelegate.cs

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
## Status
**Ready**

## Fixes or Implements [AB#8108](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8108)
- [ ] Enhancement
- [x] Bug
- [ ] Documentation

## Description:
Fixes the count of logged in users today in admin module.

## Testing
- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

### Steps to Reproduce:
Login to admin module.
Logged in users today always show 0.

### UI Changes
NO

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_403)

File: Apps/Database/src/Delegates/DBProfileDelegate.cs [hunk 1]
```diff
@@ -161,7 +161,7 @@ public int GeUnregisteredInvitedUsersCount()
         public int GetLoggedInUsersCount()
```
Comment (FuriousLlama) [inline Apps/Database/src/Delegates/DBProfileDelegate.cs]: Name is a tad misleading. How about something like GetTodayLogedinUserCount

File: Apps/Database/src/Delegates/DBProfileDelegate.cs [hunk 2]
```diff
@@ -161,7 +161,7 @@ public int GeUnregisteredInvitedUsersCount()
...
         {
             int result = this.dbContext.UserProfile
-                .Count(u => u.LastLoginDateTime >= DateTime.Today.ToUniversalTime());
+                .Count(u => u.LastLoginDateTime.HasValue && u.LastLoginDateTime.Value >= DateTime.UtcNow.Date);
```
Comment (FuriousLlama) [inline Apps/Database/src/Delegates/DBProfileDelegate.cs]: Why greater or equal? Also, the ufc offset could count users from the previous night as logged in today and the ones in the afternoon as counted for tomorrow.

Comment (tiagograf-AOT) [reply Apps/Database/src/Delegates/DBProfileDelegate.cs]: If the last login is recorded as utc there shouldnt be an issue, query should be:

Comment (tiagograf-AOT) [reply Apps/Database/src/Delegates/DBProfileDelegate.cs]: LastLoginDateTime (2020-03-27 13:00) >= Current Date (2020-03-27 00:00) both converted to utc

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_403)

File: Apps/Database/src/Delegates/DBProfileDelegate.cs (hunk 2 above)
Comment (FuriousLlama) [reply Apps/Database/src/Delegates/DBProfileDelegate.cs]: On UTC the comparison is correct. However the admin that needs to see if it is on the same day on its own time zone.

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_403)

File: Apps/Database/src/Delegates/DBProfileDelegate.cs [hunk 3]
```diff
@@ -158,10 +158,10 @@ public int GeUnregisteredInvitedUsersCount()
         }
 
         /// <inheritdoc />
-        public int GetTodayLoggedinUsersCount()
+        public int GetLoggedInUsersCount(DateTime startDate)
         {
             int result = this.dbContext.UserProfile
-                .Count(u => u.LastLoginDateTime.HasValue && u.LastLoginDateTime.Value >= DateTime.UtcNow.Date);
+                .Count(u => u.LastLoginDateTime.HasValue && u.LastLoginDateTime.Value >= startDate);
```
Comment (FuriousLlama) [inline Apps/Database/src/Delegates/DBProfileDelegate.cs]: Not sure this solves the problem, Instead how about converting both dates to a passed timeZone. That way the calculation can still be made and know for sure that the date will match

Comment (tiagograf-AOT) [reply Apps/Database/src/Delegates/DBProfileDelegate.cs]: Then the database delegate would be handling date logic which i think its wrong. 
I'm pretty sure this should work though, the utc today is converted to vancouver today, and only utc records logged in after that will make to the count. using the scenario we talked about today would be 2020-03-30 08:00:00 utc

Comment (FuriousLlama) [reply Apps/Database/src/Delegates/DBProfileDelegate.cs]: The issue is that the delegate needs to do the query, which needs to account for the time zone. To make the delegate agnostic we can pass the timezone as a paramater

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_403)

File: Apps/Database/src/Delegates/DBProfileDelegate.cs [hunk 4]
```diff
@@ -158,10 +158,12 @@ public int GeUnregisteredInvitedUsersCount()
         }
 
         /// <inheritdoc />
-        public int GetLoggedInUsersCount()
+        public int GetTodayLoggedInUsersCount(TimeZoneInfo timezone)
         {
+            DateTime startDate = TimeZoneInfo.ConvertTimeFromUtc(DateTime.UtcNow.Date, timezone);
```
Comment (FuriousLlama) [inline Apps/Database/src/Delegates/DBProfileDelegate.cs]: nit: startDate =? today

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_403)

File: Apps/Database/src/Delegates/DBProfileDelegate.cs (hunk 4 above)
Comment (bradhead) [reply Apps/Database/src/Delegates/DBProfileDelegate.cs]: nit2: startDate => today => todaysDate

**Pull request merged**